File: easy/1189_maximum_number_of_ballons/solution.cpp
#include <string>
#include <unordered_map>

class Solution {
public:
  int maxNumberOfBalloons(std::string text) {
    std::unordered_map<char, int> textCounter;

    for (const char &ch : text) {
      ++textCounter[ch];
    }

    return std::min({textCounter['b'], textCounter['a'], textCounter['l'] / 2,
                     textCounter['o'] / 2, textCounter['n']});
  }
};

File: easy/1189_maximum_number_of_ballons/solution.py
from collections import Counter


class Solution:
    def maxNumberOfBalloons(self, text: str) -> int:
        text_counter = Counter(text)

        return min(
            text_counter.get("b", 0),
            text_counter.get("a", 0),
            text_counter.get("l", 0) // 2,
            text_counter.get("o", 0) // 2,
            text_counter.get("n", 0),
        )
